src/store/transactions/deposit/deposit.thunks.ts

```diff
@@ -0,0 +1,251 @@
+                dispatch(depositActions.loadEthereumAccountSuccess(ethereumAccount));
+              } else {
+                dispatch(depositActions.loadEthereumAccountFailure("Ethereum account not found"));
+              }
+            })
+            .catch((error) => {
+              const errorMsg =
+                error instanceof Error
+                  ? error.message
+                  : "Oops ... There was an error fetching the ethereum account";
+              dispatch(depositActions.loadEthereumAccountFailure(errorMsg));
+            });
+        }
+      );
+    }
+  };
+}
+
+/**
+ * Fetches the transactions which are in the transactions pool
+ * @returns {void}
+ */
+function fetchPoolTransactions(): AppThunk {
+  return (dispatch: AppDispatch, getState: () => AppState) => {
+    dispatch(depositActions.loadPoolTransactions());
+
+    const {
+      global: { wallet },
+    } = getState();
+
+    if (wallet !== undefined) {
+      getPoolTransactions(undefined, wallet.publicKeyCompressedHex)
+        .then((transactions) => dispatch(depositActions.loadPoolTransactionsSuccess(transactions)))
+        .catch((err) => dispatch(depositActions.loadPoolTransactionsFailure(err)));
+    }
+  };
+}
+
+/**
+ * Fetches the accounts to use in the transaction on Ethereum
+ */
+function fetchAccounts(fiatExchangeRates: FiatExchangeRates, preferredCurrency: string): AppThunk {
+  return (dispatch: AppDispatch, getState: () => AppState) => {
+    const {
+      global: { wallet, tokensPriceTask },
+    } = getState();
+
+    dispatch(depositActions.loadEthereumAccounts());
+
+    if (wallet !== undefined) {
+      return CoordinatorAPI.getTokens(undefined, undefined, undefined, undefined, 2049).then(
+        (res) => {
+          ethereum
+            .getTokens(wallet, res.tokens)
+            .then((ethereumAccounts): EthereumAccount[] => {
+              return ethereumAccounts.map((ethereumAccount) => {
+                const tokenFromTokensPrice =
+                  tokensPriceTask.status === "successful" &&
+                  tokensPriceTask.data[ethereumAccount.token.id];
+                const token = tokenFromTokensPrice ? tokenFromTokensPrice : ethereumAccount.token;
+
+                const fiatBalance = convertTokenAmountToFiat(
+                  ethereumAccount.balance,
+                  token,
+                  preferredCurrency,
+                  fiatExchangeRates
+                );
+
+                return { ...ethereumAccount, balance: ethereumAccount.balance, fiatBalance };
+              });
+            })
+            .then((metaMaskTokens) =>
+              dispatch(depositActions.loadEthereumAccountsSuccess(metaMaskTokens))
+            )
+            .catch((err) => dispatch(depositActions.loadEthereumAccountsFailure(err)));
+        }
+      );
+    }
+  };
+}
+
+/**
+ * Fetches the recommended fees from the Coordinator
+ * @returns {void}
+ */
+function fetchFees(): AppThunk {
```

Comment:
Yes onLoadFees is just used on the view when is a Exit

Reply:
Interesting!

> Yes onLoadFees is just used on the view when is a Exit

This is not strictly true:

![image](https://user-images.githubusercontent.com/15896177/140744387-9fc7cd43-c2c7-420d-99f2-0e6335972386.png)

Can I then move `onLoadFees()` to the `if` so it's only executed when `transactionType === TxType.Exit`?

Otherwise, I'm forced to pass some dummy function, which is ugly.

Reply:
No! Fees are fetched for L2 txs only, so we need to call the `onLoadFees` for transactions and for exits.

Reply:
Done!